### PR TITLE
Add new FAOSTAT variables

### PR DESCRIPTION
**Pull request by pabloarosado**
This PR adds a FAOSTAT garden dataset (and its corresponding grapher dataset) that contains various tables. Each table replaces a grapher dataset that was created using an old version of FAOSTAT data (disconnected from our current pipeline), namely:
* [Arable land per crop output (PIN) (FAO, 2019)](https://owid.cloud/admin/datasets/4248) -> Table `arable_land_per_crop_output`.
* [Global agricultural land by crop - FAO (2017)](https://owid.cloud/admin/datasets/665) -> Table `area_used_per_crop_type`.
* [Share of fish stocks within biologically sustainable levels (FAO, 2020)](https://owid.cloud/admin/datasets/5281) -> Table `share_of_sustainable_and_overexploited_fish`.
* [Land sparing due to crop yield increases (UN FAO, 2021)](https://owid.cloud/admin/datasets/5340) -> Table `land_spared_by_increased_crop_yields`.
* [Dietary macronutrient compositions - FAO (2017)](https://owid.cloud/admin/datasets/654) -> Table `food_available_for_consumption`.
* [Diet compositions by commodity categories - FAO (2017)](https://owid.cloud/admin/datasets/658) -> Table `macronutrient_compositions`.

This dataset is quite unusual since it contains tables of different topics in the same dataset. But it's built this way simply for convenience because those tables contain mostly one or a few variables. If we need to add any other FAOSTAT variable in the future, we can continue using this dataset.

### CODE REVIEW

File: etl/steps/data/garden/faostat/2023-02-22/additional_variables.py [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+    return tb_spared_land
+
+
+def generate_food_available_for_consumption(df_fbsc: pd.DataFrame) -> Table:
+    # Element code for "Food available for consumption" of faostat_fbsc (in kilocalories per day per capita).
+    ELEMENT_CODE_FOR_PER_CAPITA_FOOD = "0664pc"
+    # Expected unit.
+    CONSUMPTION_UNIT = "kilocalories per day per capita"
+
+    # df_fbsc[df_fbsc["unit"].str.contains("kilocal")][["element_code", "element", "unit"]].drop_duplicates()
```
Comment (paarriagadap) [inline etl/steps/data/garden/faostat/2023-02-22/additional_variables.py:291]: You could maybe delete this comment if you are not using it anymore

**Review by paarriagadap — APPROVED** [review]
Hi @pabloarosado, looks great to me! I only added some minor comments

File: etl/steps/data/garden/faostat/2023-02-22/additional_variables.meta.yml [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+tables:
+  arable_land_per_crop_output:
+    variables:
+      area:
+        title: 'Arable land'
+        unit: 'hectares'
+        short_unit: 'ha'
+        description: |
+          Arable land is the total of areas (extent of surface of land or water) under temporary crops, temporary meadows and pastures, and land with temporary fallow.  Arable land does not include land that is potentially cultivable but is not normally cultivated.
+      index:
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply etl/steps/data/garden/faostat/2023-02-22/additional_variables.meta.yml:26]: Good suggestion, but I'm not sure if there are going to be new charts from this data, so it won't be of help anymore. Also, if it's avoidable, I prefer not to mention `grapher` metadata in `garden` steps (although I agree that it's often convenient).

File: etl/steps/data/garden/faostat/2023-02-22/additional_variables.meta.yml [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+      energy_from_animal_products:
+        title: "Daily caloric intake per person from animal products"
+        unit: "kilocalories per day per capita"
+        short_unit: "kcal"
+        description: &macronutrient_composition_variable_description |
+          The FAO provide annual figures from 1961 by country on daily caloric supply, fat supply (in grams),   and protein supply (in grams). To calculate the daily per capita supply of carbohydrates, we assume an energy density by macronutrient of 4 kcal per gram of both protein and carbohydrate and 9 kcal per gram of fat (based on established nutritional guidelines reported by the FAO). The daily supply of carbohydrates is therefore calculated as:
+
+          ((Daily supply of kcal)-(Daily supply of protein * 4 + Daily supply of fat * 9)) / 4
+
+          The quantity of calories from each macronutrient is then calculated based on the energy density figures given above (e.g. calories from protein is calculated by multiplting the daily supply of protein in grams by 4).
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply etl/steps/data/garden/faostat/2023-02-22/additional_variables.meta.yml:107]: Nice catch!

File: etl/steps/data/garden/faostat/2023-02-22/additional_variables.py (hunk 1 above)
Comment (pabloarosado) [reply etl/steps/data/garden/faostat/2023-02-22/additional_variables.py:291]: Ups, true!

File: etl/steps/data/garden/faostat/2023-02-22/additional_variables.py [hunk 4]
```diff
@@ -0,0 +1,707 @@
...
+        (df_qi["element_code"] == ELEMENT_CODE_PRODUCTION_INDEX) & (df_qi["item_code"] == ITEM_CODE_FOR_CROPS)
+    ].reset_index(drop=True)
+    # Sanity check.
+    error = "Unit for element 'Gross Production Index Number (2014-2016 = 100)' in faostat_qi has changed."
+    assert list(df_qi["unit"].unique()) == ["index"], error
+    # Rename columns and select only necessary columns.
+    df_qi = df_qi[["country", "year", "value"]].rename(columns={"value": "index"})
+
+    # Combine both dataframes.
+    combined = pd.merge(df_rl, df_qi, on=["country", "year"], how="inner")
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply etl/steps/data/garden/faostat/2023-02-22/additional_variables.py:50]: Good suggestion, done!

**Comment by pabloarosado** [comment]
Thanks @paarriagadap. Waiting for [this other PR](https://github.com/owid/etl/pull/997) before merging.

**Pull request merged**